cypress/e2e/register/email.spec.ts

```diff
@@ -0,0 +1,92 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/// <reference types="cypress" />
+
+import { HomeserverInstance } from "../../plugins/utils/homeserver";
+import { Mailhog } from "../../support/mailhog";
+
+describe("Email Registration", () => {
+    let homeserver: HomeserverInstance;
+    let mailhog: Mailhog;
+
+    beforeEach(() => {
+        cy.startMailhog().then((_mailhog) => {
+            mailhog = _mailhog;
+            cy.startHomeserver({
+                template: "email",
+                variables: {
+                    SMTP_HOST: "host.docker.internal",
+                    SMTP_PORT: _mailhog.instance.smtpPort,
+                },
+            }).then((_homeserver) => {
+                homeserver = _homeserver;
+
+                cy.intercept(
+                    { method: "GET", pathname: "/config.json" },
+                    {
+                        body: {
+                            default_server_config: {
+                                "m.homeserver": {
+                                    base_url: homeserver.baseUrl,
+                                },
+                                "m.identity_server": {
+                                    base_url: "https://server.invalid",
+                                },
+                            },
+                        },
+                    },
+                );
+                cy.visit("/#/register");
+                cy.injectAxe();
+            });
+        });
+    });
+
+    afterEach(() => {
+        cy.stopHomeserver(homeserver);
+        cy.stopMailhog(mailhog);
+    });
+
+    it("registers an account and lands on the use case selection screen", () => {
+        cy.findByRole("textbox", { name: "Username" }).should("be.visible");
+        // Hide the server text as it contains the randomly allocated Homeserver port
+        const percyCSS = ".mx_ServerPicker_server { visibility: hidden !important; }";
+
+        cy.findByRole("textbox", { name: "Username" }).type("alice");
+        cy.findByPlaceholderText("Password").type("totally a great password");
+        cy.findByPlaceholderText("Confirm password").type("totally a great password");
+        cy.findByPlaceholderText("Email").type("alice@email.com");
+        cy.findByRole("button", { name: "Register" }).click();
+
+        cy.findByText("Check your email to continue").should("be.visible");
+        cy.percySnapshot("Registration check your email", { percyCSS });
+        cy.checkA11y();
+
+        cy.findByText("An error was encountered when sending the email").should("not.exist");
+
+        // Unfortunately the email is not available immediately, so we have a magic wait here
+        cy.wait(5000).then(async () => {
```

Comment:
Can we do a waitFor or similar here, to poll (maybe every 1000ms) instead of a single wait? This would be a) quicker if the email arrives quicker and b) more robust if the email server is slow for some reason.